src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -11,26 +11,44 @@ namespace Microsoft.Extensions.Configuration
     /// </summary>
     public static class ConfigurationPath
     {
+        /// <summary>
+        /// The deliminiter used for keys is ":".
+        /// </summary>
         public static readonly string KeyDelimiter = ":";
 
-        public static string Combine(params string[] pathSegements)
+        /// <summary>
+        /// Combines all several path segments into one path.
+        /// </summary>
+        /// <param name="pathSegments">The path segments to combine.</param>
+        /// <returns>The combined path.</returns>
+        public static string Combine(params string[] pathSegments)
         {
-            if (pathSegements == null)
+            if (pathSegments == null)
             {
-                throw new ArgumentNullException(nameof(pathSegements));
+                throw new ArgumentNullException(nameof(pathSegments));
             }
-            return string.Join(KeyDelimiter, pathSegements);
+            return string.Join(KeyDelimiter, pathSegments);
         }
 
-        public static string Combine(IEnumerable<string> pathSegements)
+        /// <summary>
+        /// Combines all several path segments into one path.
+        /// </summary>
+        /// <param name="pathSegments">The path segments to combine.</param>
+        /// <returns>The combined path.</returns>
+        public static string Combine(IEnumerable<string> pathSegments)
         {
-            if (pathSegements == null)
+            if (pathSegments == null)
             {
-                throw new ArgumentNullException(nameof(pathSegements));
+                throw new ArgumentNullException(nameof(pathSegments));
             }
-            return string.Join(KeyDelimiter, pathSegements);
+            return string.Join(KeyDelimiter, pathSegments);
         }
 
+        /// <summary>
+        /// Extracts the section key from a path.
```

Comment:
Erm, feel free to file another issue for the rename, but I definitely don't think we should do it as part of a doc comments PR :)